frontend/src/components/worksheets/BundleDetail/BundleDetail.jsx

```diff
@@ -1,250 +1,215 @@
+     * @return  jQuery Deferred object
+     */
+    const fetchFileSummary = (uuid, path)=> {
+        return $.ajax({
+            type: 'GET',
+            url: '/rest/bundles/' + uuid + '/contents/blob' + path,
+            data: {
+                head: 50,
+                tail: 50,
+                truncation_text: '\n... [truncated] ...\n\n',
+            },
+            dataType: 'text',
+            cache: false,
+            context: this, // automatically bind `this` in all callbacks
+        });
+    }
+
+    const fetcherMetadata = (url) =>
+        fetch(url, {
+            type: 'GET',
+            url: url,
+            dataType: 'json',
+        })
+            .then((r) => {
+                return r.json();
+            })
+            .catch((error) => {
+                setBundleInfo(null);
+                setFileContents(null);
+                setStderr(null);
+                setStdout(null);
+                setErrorMessages(errorMessages=>errorMessages.concat([error]))
+            });
+
+    const urlMetadata =
+        '/rest/bundles/' + uuid+  "?" + new URLSearchParams({include_display_metadata: 1,include: 'owner,group_permissions,host_worksheets'}).toString()
+
+    const{dataMetadata,errorMetadata,mutateMetadata} = useSWR(urlMetadata, fetcherMetadata, {
+        revalidateOnMount: true,
+        refreshInterval:refreshInterval,
+        onSuccess: (response, key, config) => {
+            // Normalize JSON API doc into simpler object
+            const bundleInfo = new JsonApiDataStore().sync(response);
+            bundleInfo.editableMetadataFields = response.data.meta.editable_metadata_keys;
+            bundleInfo.metadataType = response.data.meta.metadata_type;
+            setBundleInfo( bundleInfo );
+        }
+    });
+
+    const fetcherContents = (url) =>
+        $.ajax( {
+            type: 'GET',
+            url: url,
+            dataType: 'json',
+        }).fail(
+                function(xhr, status, err) {
+                // 404 Not Found errors are normal if contents aren't available yet, so ignore them
+                if (xhr.status !== 404) {
+                    setBundleInfo(null);
+                    setFileContents(null);
+                    setStderr(null);
+                    setStdout(null);
+                    setErrorMessages(errorMessages=>errorMessages.concat([xhr.responseText]));
+                } else {
+                    // If contents aren't available yet, then also clear stdout and stderr.
+                    setFileContents(null);
+                    setStderr(null);
+                    setStdout(null);
+                }
+            }
+            );
+
+    const urlContents =
+        '/rest/bundles/' + uuid + '/contents/info/' +  "?" + new URLSearchParams({depth: 1});
+
+    useSWR(urlContents, fetcherContents, {
+        revalidateOnMount: true,
+        refreshInterval:refreshInterval,
+        onSuccess: (response, key, config) => {
```

Comment:
Can we refactor this into a separate function so it's cleaner?

Reply:
Do you mean refactor the onSuccess callback function?

Reply:
Yeah

Reply:
solved :)